### ISSUE TITLE
Comprehensive GitHub Actions CI config for SIMD testing/benchmarking

### ISSUE THREAD

**Issue by chessai** [emilypi/base64#51]
We could take a page out of https://raw.githubusercontent.com/haskell/bytestring/master/.github/workflows/ci.yml

**Comment by emilypi** [comment]
Makes sense to me. Haskell-ci seems to be failing to produce useful CI at this point.
